ManagementSystem/PhoneBook.js

```diff
@@ -0,0 +1,93 @@
+const fs = require('fs');
+const PersonalContact = require('./PersonalContact');
+const BusinessContact = require('./BusinessContact');
+const { ContactNotFoundError, FileSystemError } = require('./Errors');
+
+const fileName = 'contacts.txt';
+
+class PhoneBook {
+    constructor() {
+        this.contacts = [];
+        this.loadContactsFromFile();
+    }
+
+    addContact(contact) {
+        this.contacts.push(contact);
+        this.saveContactsToFile();
+    }
+
+    updateContact(name, newContact) {
+        const index = this.contacts.findIndex(contact => contact.name === name);
+        if (index !== -1) {
+            this.contacts[index] = newContact;
+            this.saveContactsToFile();
+        } else {
+            throw new ContactNotFoundError(`Contact name "${name}" not found.`);
+        }
+    }
+
+    deleteContact(name) {
+        const index = this.contacts.findIndex(contact => contact.name === name);
+        if (index !== -1) {
+            this.contacts.splice(index, 1);
+            this.saveContactsToFile();
```

Comment:
If we are deleting contacts as per the function then why we are calling **saveContactsToFile** here?

Reply:
Each time a contact is deleted, I want to make sure the file reflets the changes, if I didn't call saveContactToFile, the contact will be removed from the memory, but still exist in the file.

Reply:
Got it, Then maybe please look for a better name for this method as it is generic for saving a file or deleting a file.

Reply:
Got it, Then maybe please look for a better name for this method as it is generic for saving a file or deleting a file.